yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,130 @@
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Inventory/hardware path of the FRU whose keyword needs to be
+                updated.
+          - name: data
+            type:
+                variant[struct[string,string,array[byte]],
+                struct[string,array[byte]]]
+            description: >
+                Contains data required to write. If it is of IPZ type VPD the
+                data should be passed in tuple[record, keyword, value] format
+                and if it is of keyword type VPD the data will be passed in
+                tuple[keyword, data] format, etc. The format of the data will be
+                depend on the type of VPD.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: ReadKeyword
+      description: >
+          Method to read VPD keyword value. Keyword can be read from hardware or
+          from cache.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Inventory/hardware Path of the FRU whose keyword needs to be
+                read.
+          - name: data
+            type: variant[struct[string,string],struct[string]]
+            description: >
+                Contains data requied to read. If it is of IPZ type VPD the data
+                should be passed in tuple[record, keyword] format and if it is
+                of keyword type VPD the data will be passed in tuple[keyword]
+                format, etc. The format of the data will be depend on the type
+                of VPD.
+          - name: target
+            type: size
+            description: >
+                Target takes 0/1 to read keyword from cache/hardware
+                respectively.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                Value of the given keyword read from the given target.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: GetExpandedLocationCode
+      description: >
+          Method to get the expanded location code for the given inventory path.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                Location code in expanded format.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: deleteFRUVPD
+      description: >
+          Method to delete given FRU VPD and set its present property as false.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Invnetory path of the FRU.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: CollectFRUVPD
```

Comment:
this function will be available only if JSON define is used that also needs to be documented

Reply:
updated